### DISCUSSION TITLE
How can I output the POM element locator string in WebdriverIO?

### DISCUSSION THREAD

**Discussion by darkartswizard** [webdriverio/webdriverio#7549] [Q&A] [answered]
Let's say I have an element in a Page / Object class
```
get btnSkip() { return browser.$('#button-skip') }
```

If the element exists I can pull the full description from the DOM
```
                    locatorHtml = btnSkip.getHTML();
                    this.log(`Element innerHTML : ${locatorHtml}`);
```

But if the element did not exist on the webpage, 
I would like to output the locator description from the original POM element.

I tried this but no string was returned.
```
console.log (btnSkip.toString()) // output "#button-skip"
console.log (btnSkip.toString) // output "#button-skip"
```
Any thoughts on what I'm doing wrong?

Thank you.

**Comment by wswebcreation** [comment, marked as answer]
Not sure how you've set up your project but you can do something like this

```js
console.log('selector = ', (await $('#foo')).selector)
# output:  [0-0] selector =  #foo
```

**Reply by darkartswizard** [reply]
That is exactly the answer I needed!
Thank you!

**Comment by erwinheitzman** [comment]
I agree with what @wswebcreation said and would like to add that you can also combine it with `console.assert` like so:
```js
// if the condition is falsy, it will log the selector. if it is truthy, it does not log anything
console.assert(element.isExisting(), `element with selector ${element.selector} did not exist`)
```

Example of output:
```
Assertion failed: element with selector #foo did not exist
```

**Reply by darkartswizard** [reply]
Thanks. I'm doing something a little more advanced. 
I'm adding Dynamic Element Class and Locator Switching.
It outputs the locator description that is no longer relevant, then makes three attempts to build a valid locator.
That way the framework outputs the elements that need to be maitained in the Page Object Classes, but has a better shot at still completing the journey.
